src/main/java/de/picturesafe/search/elasticsearch/connect/query/preprocessor/StandardQuerystringPreprocessor.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2020 picturesafe media/data/bank GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.picturesafe.search.elasticsearch.connect.query.preprocessor;
+
+import de.picturesafe.search.elasticsearch.config.QueryConfiguration;
+import de.picturesafe.search.elasticsearch.connect.query.QuerystringPreprocessor;
+import org.apache.commons.lang3.StringUtils;
+import org.elasticsearch.index.query.Operator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+public class StandardQuerystringPreprocessor implements QuerystringPreprocessor {
+
+    static final String PHRASE_DELIMITER = "\"";
+    static final String ESCAPE_CHARACTER = "\\";
+    static final Locale REPLACEMENT_LOCALE = Locale.GERMAN;
+
+    static final String TOKEN_OR = "||";
+    static final String TOKEN_AND = "&&";
+    static final String TOKEN_NOT = "NOT";
+    static final String TOKEN_NEAR_BY = "~";
+
+    private static final String DEFAULT_TOKEN_DELIMITERS = " ,\"(){}[]:=\\/^~";
+    private static final boolean DEFAULT_AUTO_BRACKET = true;
+    private static final boolean DEFAULT_INSERT_MISSING_OPERATORS = true;
+    private static final Map<String, String> DEFAULT_REPLACEMENTS;
+
+    private String tokenDelimiters = DEFAULT_TOKEN_DELIMITERS;
+    private Map<String, String> replacements = DEFAULT_REPLACEMENTS;
+    private boolean autoBracket = DEFAULT_AUTO_BRACKET;
+    private boolean insertMissingOperators = DEFAULT_INSERT_MISSING_OPERATORS;
+
+    private final OptimizerAutoBracket queryOptimizerAutoBracket = new OptimizerAutoBracket();
+    private final OptimizerDefaultOperator queryOptimizerDefaultOperator;
+
+    static {
+        DEFAULT_REPLACEMENTS = new HashMap<>();
+        DEFAULT_REPLACEMENTS.put(",", " " + TOKEN_OR + " ");  // Blanks are necessary, do not remove!
+        DEFAULT_REPLACEMENTS.put("|", TOKEN_OR);
+        DEFAULT_REPLACEMENTS.put("oder", TOKEN_OR);
+        DEFAULT_REPLACEMENTS.put("or", TOKEN_OR);
+        DEFAULT_REPLACEMENTS.put("&", TOKEN_AND);
+        DEFAULT_REPLACEMENTS.put("+", TOKEN_AND);
+        DEFAULT_REPLACEMENTS.put("und", TOKEN_AND);
+        DEFAULT_REPLACEMENTS.put("and", TOKEN_AND);
```

Comment:
I would provide a configuration option (also to support more languages), for example in elasticsearch.properties something like:
standard_querystring_preprocessor.replacement_tokens.and=& + and und

Reply:
done